src/components/common/Input/index.tsx

```diff
@@ -2,30 +2,47 @@ import classNames from 'classnames';
 import { ChangeEvent, useEffect, useState, useRef, SetStateAction, Dispatch } from 'react';
 import $ from '@/components/common/Input/Input.module.scss';
 import X from '@/assets/svg/X.svg?react';
+
 interface InputProps {
   text: string;
   setText: Dispatch<SetStateAction<string>>;
   maxLength?: number;
+  inputMode?: 'text' | 'numeric';
 }
 
-export default function Input({ text, setText, maxLength }: InputProps) {
+export default function Input({ text, setText, maxLength, inputMode }: InputProps) {
   const [isTyping, setIsTyping] = useState<boolean>(false);
-  //input을 focus하기 위해 useRef 사용(handleClear 실행 이후에도 포커스가 유지되도록)
-  const inputRef = useRef<HTMLInputElement>(null);
+  //textarea을 focus하기 위해 useRef 사용(handleClear 실행 이후에도 포커스가 유지되도록)
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
   //div 밖 영역 감지를 위해
   const inputWrapperRef = useRef<HTMLDivElement>(null);
 
-  const onChangeText = (e: ChangeEvent<HTMLInputElement>) => {
+
+  // 장문 입력 시 textarea 높이를 자동으로 조절함 
+  const handleResizeHeight = () => {
+    const textarea = textareaRef.current;
+    if (textarea) {
+      textarea.style.height = 'auto';
+      const newHeight = textarea.scrollHeight; 
+      textarea.style.height = `${newHeight}px`; 
+    }
```

Comment:
이렇게 하면 가독성이 더 좋을거같아요!
```suggestion
  const textarea = textareaRef.current;
  if (!textarea) return;

  textarea.style.height = 'auto';
  textarea.style.height = `${textarea.scrollHeight}px`;
   }
```

Reply:
아하 불필요한 변수를 줄이니 더 깔끔해지네요! 반영하겠습니다~!